apps/src/applab/applab.js

```diff
@@ -1246,6 +1255,54 @@ Applab.onDivApplabClick = function (event) {
   Applab.editElementProperties(event.target);
 };
 
+/**
+ * @param el {Element}
+ * @returns {number} The outerWidth (width + margin) of the element in pixels,
+ * or NaN if element's css width or margin are not defined.
+ */
+Applab.getOuterWidth = function(el) {
+  var marginLeft = parseInt($(el).css('margin-left'));
+  var marginRight = parseInt($(el).css('margin-right'));
+  return parseInt(el.style.width) + marginLeft + marginRight;
```

Comment:
jquery's outerWidth getter is capable of reporting the outer width including margin, however the setter cannot be made to include margin. I considered using jQuery's outerWidth for the getter only, but that seems risky since then any inconsistency between the way jQuery and I compute outer width on any browser would result in bad behavior.